components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/internal/OAuth2ServiceComponent.java

```diff
@@ -834,7 +835,35 @@ protected void unsetOAuth2RevocationProcessor(OAuth2RevocationProcessor oAuth2Re
         if (log.isDebugEnabled()) {
             log.debug("Unset Oauth2 revocation processor.");
         }
-        OAuth2ServiceComponentHolder.setOrganizationUserResidentResolverService(null);
+        OAuth2ServiceComponentHolder.getInstance().setRevocationProcessor(null);
+    }
+
+    @Reference(
+            name = "oauth2.token.persistence.handler",
+            service = IndirectTokenRevocationHandler.class,
+            cardinality = ReferenceCardinality.OPTIONAL,
```

Comment:
Cardinality optional means that the system can run without a IndirectTokenRevocationHandler. Is that the expectation?